lantern_embeddings_core/src/lib.rs

```diff
@@ -394,10 +396,115 @@ pub mod clip {
         Ok(())
     }
 
+    fn clear_model_cache(
+        model_map: &mut HashMap<&'static str, ModelInfo>,
+    ) -> Result<(), anyhow::Error> {
+        for (_, model_info) in model_map.iter_mut() {
+            model_info.encoder = None;
+        }
+
+        Ok(())
+    }
+
+    fn check_available_gpu_memory(mem_threshold: f64) -> Result<bool, anyhow::Error> {
```

Comment:
Can you change the name of this function () so the meaning of its return is clearer?
Or, maybe change this into `percent_gpu_memory_available() -> Result<f64, Error>`